src/nvim/os/fs.c

```diff
@@ -148,11 +148,7 @@ int os_nodetype(const char *name)
     case UV_TCP:         // unix only
     case UV_UNKNOWN_HANDLE:
     default:
-#ifdef WIN32
-      nodetype = NODE_OTHER;
```

Comment:
This came from os_win32.c in Vim. Most likely we need to also bring over win32-specific parts of `buf_write` and `readfile` to match the logic from Vim, where it makes sense to do so.

I'm working on this now.